lib/auth/default_current_user_provider.rb

```diff
@@ -290,14 +320,19 @@ def is_user_api?
   end
 
   def has_auth_cookie?
-    cookie = @request.cookies[TOKEN_COOKIE]
-    !cookie.nil? && cookie.length == 32
+    cookie_string = @request.cookies[TOKEN_COOKIE]
+    return false if cookie_string.nil?
```

Comment:
```suggestion
    return false if cookie_string.blank?
```

`blank?` is probably better when dealing with strings.